http.go

```diff
@@ -120,6 +120,8 @@ type Response struct {
 	raddr net.Addr
 	// Local TCPAddr from concurrently net.Conn
 	laddr net.Addr
+
+	headersWritten bool
```

Comment:
I don't think this state needs to be kept here, you can just add a `headersWritten bool` to `UnbufferedWriterHttp1`.